clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -178,6 +178,11 @@ public CommitRequestManager(
      */
     @Override
     public NetworkClientDelegate.PollResult poll(final long currentTimeMs) {
+        if (coordinatorRequestManager.coordinator().isEmpty() && closing) {
+            handleClosingWithoutCoordinator();
+            return EMPTY;
+        }
+
         // poll when the coordinator node is known and fatal error is not present
         if (coordinatorRequestManager.coordinator().isEmpty()) {
             pendingRequests.maybeFailOnCoordinatorFatalError();
```

Comment:
Perhaps it could be written like this for better readability:
```suggestion
            pendingRequests.maybeFailOnCoordinatorFatalError();
            if(closing && pendingRequests.hasUnsentRequests()){
                CommitFailedException exception = new CommitFailedException(
                "Failed to commit offsets: Coordinator unknown and consumer is closing");
                pendingRequests.drainPendingCommits()
                  .forEach(request -> request.future().completeExceptionally(exception));
            }
```